syntax ="proto3";
package simple;
message Request{
    string query =1;
}

message Response{
    string response =1;
}

message HealthCheckRequest {
    string service = 1;
  }
   
  message HealthCheckResponse {
    enum ServingStatus {
      UNKNOWN = 0;
      SERVING = 1;
      NOT_SERVING = 2;
      SERVICE_UNKNOWN = 3;  // Used only by the Watch method.
    }
    ServingStatus status = 1;
  }
   
service RequestService{

    rpc rService(Request)returns (Response){}
    rpc Check(HealthCheckRequest) returns (HealthCheckResponse){};
 
    rpc Watch(HealthCheckRequest) returns (stream HealthCheckResponse){};
}